src/messages.rs

```diff
@@ -94,29 +106,46 @@ impl ErrorReturn {
 }
 
 /// These are the messageTypes routing provides
-/// many are internal to routing and woudl not be useful
-/// to users.
 #[derive(PartialEq, Eq, Clone, Debug, RustcEncodable, RustcDecodable)]
-pub enum MessageType {
-    ConnectRequest(ConnectRequest),
-    ConnectResponse(ConnectResponse),
+pub enum Request {
+    Get(DataRequest),
+    Put(Data),
+    Post(Data),
+    Delete(DataRequest),
+    Refresh(u64, Vec<u8>),
+}
+
+#[derive(PartialEq, Eq, Clone, Debug, RustcEncodable, RustcDecodable)]
+pub enum Response {
+    Get(GetDataResponse),
+    Put(ErrorReturn),
+    Post(ErrorReturn),
+    Delete(ErrorReturn),
+}
+
+#[derive(PartialEq, Eq, Clone, Debug, RustcEncodable, RustcDecodable)]
+pub enum InternalRequest {
+    Connect(ConnectRequest),
     FindGroup,
-    FindGroupResponse(Vec<PublicId>),
-    GetData(DataRequest),
-    GetDataResponse(GetDataResponse),
-    DeleteData(DataRequest),
-    DeleteDataResponse(ErrorReturn),
     GetGroupKey,
-    GetGroupKeyResponse(BTreeMap<NameType, sign::PublicKey>),
-    Post(Data),
-    PostResponse(ErrorReturn, BTreeMap<NameType, sign::PublicKey>),
-    PutData(Data),
-    PutDataResponse(ErrorReturn, BTreeMap<NameType, sign::PublicKey>),
     PutKey,
     PutPublicId(PublicId),
-    PutPublicIdResponse(PublicId, SignedMessage),
-    Refresh(u64, Vec<u8>),
-    Unknown,
+}
+
+#[derive(PartialEq, Eq, Clone, Debug, RustcEncodable, RustcDecodable)]
+pub enum InternalResponse {
+    Connect(ConnectResponse),
+    FindGroup(Vec<PublicId>),
+    GetGroupKey(BTreeMap<NameType, sign::PublicKey>),
+    PutPublicId(PublicId, SignedToken),
+}
+
+#[derive(PartialEq, Eq, Clone, Debug, RustcEncodable, RustcDecodable)]
+pub enum Content {
+    Request(Request),
+    InternalRequest(InternalRequest),
+    Response(Response),
```

Comment:
Does a Response not also need a SignedToken; and is it possible that to be a full type?